extensions/statistics/src/Api/Controller/ShowStatisticsData.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Statistics\Api\Controller;
+
+use DateTime;
+use DateTimeZone;
+use Flarum\Discussion\Discussion;
+use Flarum\Http\RequestUtil;
+use Flarum\Post\Post;
+use Flarum\Post\RegisteredTypesScope;
+use Flarum\Settings\SettingsRepositoryInterface;
+use Flarum\User\User;
+use Illuminate\Contracts\Cache\Store;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Arr;
+use Laminas\Diactoros\Response\JsonResponse;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+class ShowStatisticsData implements RequestHandlerInterface
+{
+    /**
+     * The amount of time to cache lifetime statistics data for in seconds.
+     */
+    public static $lifetimeStatsCacheTtl = 300;
+
+    /**
+     * The amount of time to cache timed statistics data for in seconds.
+     */
+    public static $timedStatsCacheTtl = 900;
+
+    protected $entities = [];
+
+    /**
+     * @var SettingsRepositoryInterface
+     */
+    protected $settings;
+
+    /**
+     * @var Store
+     */
+    protected $cache;
+
+    public function __construct(SettingsRepositoryInterface $settings, Store $cache)
+    {
+        $this->settings = $settings;
+        $this->cache = $cache;
+
+        $this->entities = [
+            'users' => [User::query(), 'joined_at'],
+            'discussions' => [Discussion::query(), 'created_at'],
+            'posts' => [Post::where('type', 'comment')->withoutGlobalScope(RegisteredTypesScope::class), 'created_at']
+        ];
+    }
+
+    public function handle(ServerRequestInterface $request): ResponseInterface
+    {
+        $actor = RequestUtil::getActor($request);
+
+        // Must be an admin to get statistics data -- this is only visible on the admin
+        // control panel.
+        $actor->assertAdmin();
+
+        $reportingPeriod = Arr::get($request->getQueryParams(), 'period');
+
+        return new JsonResponse($this->getResponse($reportingPeriod));
+    }
+
+    private function getResponse(?string $period): array
+    {
+        if ($period === 'lifetime') {
+            return $this->getLifetimeStatistics();
+        }
+
+        return array_merge(
+            $this->getTimedStatistics(),
+            ['timezoneOffset' => $this->getUserTimezone()->getOffset(new DateTime)]
+        );
+    }
+
+    private function getLifetimeStatistics()
+    {
+        $cachedVal = $this->cache->get('flarum-subscriptions.lifetime_stats');
+        if ($cachedVal) {
+            return $cachedVal;
+        }
+
+        $val = array_map(function ($entity) {
+            return $entity[0]->count();
+        }, $this->entities);
+
+        $this->cache->put('flarum-subscriptions.lifetime_stats', $val, static::$lifetimeStatsCacheTtl);
+
+        return $val;
```

Comment:
can we inject an `\Illuminate\Contracts\Cache\Repository::class` for the cache instead and do this:

```suggestion
        return $this->cache->remember('flarum-subscriptions.lifetime_stats', self::$lifetimeStatsCacheTtl, function () {
            return array_map(function ($entity) {
                return $entity[0]->count();
            }, $this->entities);
        });
```